sklearn/model_selection/tests/test_search.py

```diff
@@ -591,71 +599,190 @@ def test_param_sampler():
+    grid_search = GridSearchCV(SVC(), cv=n_folds, iid=False,
+                               param_grid=params)
+    grid_search.fit(X, y)
+    grid_search_iid = GridSearchCV(SVC(), cv=n_folds, iid=True,
+                                   param_grid=params)
+    grid_search_iid.fit(X, y)
+
+    # scipy.stats dists now supports `seed` but we still support scipy 0.12
+    # which doesn't support the seed. Hence the assertions in the test for
+    # random_search alone should not depend on randomization.
     n_search_iter = 30
-    search = RandomizedSearchCV(SVC(), n_iter=n_search_iter, cv=n_cv_iter,
-                                param_distributions=params, iid=False)
-    search.fit(X, y)
-    assert_equal(len(search.grid_scores_), n_search_iter)
-
-    # Check consistency of the structure of each cv_score item
-    for cv_score in search.grid_scores_:
-        assert_equal(len(cv_score.cv_validation_scores), n_cv_iter)
-        # Because we set iid to False, the mean_validation score is the
-        # mean of the fold mean scores instead of the aggregate sample-wise
-        # mean score
-        assert_almost_equal(np.mean(cv_score.cv_validation_scores),
-                            cv_score.mean_validation_score)
-        assert_equal(list(sorted(cv_score.parameters.keys())),
-                     list(sorted(params.keys())))
-
-    # Check the consistency with the best_score_ and best_params_ attributes
-    sorted_grid_scores = list(sorted(search.grid_scores_,
-                              key=lambda x: x.mean_validation_score))
-    best_score = sorted_grid_scores[-1].mean_validation_score
-    assert_equal(search.best_score_, best_score)
-
-    tied_best_params = [s.parameters for s in sorted_grid_scores
-                        if s.mean_validation_score == best_score]
-    assert_true(search.best_params_ in tied_best_params,
-                "best_params_={0} is not part of the"
-                " tied best models: {1}".format(
-                    search.best_params_, tied_best_params))
-
-
-def test_grid_search_score_consistency():
+    params = dict(C=expon(scale=10), gamma=expon(scale=0.1))
+    random_search = RandomizedSearchCV(SVC(), n_iter=n_search_iter, cv=n_folds,
+                                       iid=False, param_distributions=params)
+    random_search.fit(X, y)
+    random_search_iid = RandomizedSearchCV(SVC(), n_iter=n_search_iter,
+                                           cv=n_folds, iid=True,
+                                           param_distributions=params)
+    random_search_iid.fit(X, y)
+
+    param_keys_gs = ('param_C', 'param_degree', 'param_gamma', 'param_kernel')
+    param_keys_rs = ('param_C', 'param_gamma')
+    score_keys = ('test_accuracy_mean',
+                  'test_accuracy_rank',
+                  'test_split_0_accuracy',
+                  'test_split_1_accuracy',
+                  'test_split_2_accuracy',
+                  'test_accuracy_std')
```

Comment:
I was also wondering about the same thing... But decided to proceed with it as such because in all of the cases (`test_accuracy_mean`/ `test_split_x_accuracy`/`test_accuracy_rank`), the last word specifies what the value represents (rank/mean/std/accuracy(score))...

Thoughts @jnothman @amueller @agramfort ?